### PR TITLE
elpa-packages: build xeft dynamic module

### PR DESCRIPTION
**Pull request by acowley**
## Description of changes

Build the native dynamic module required by the xeft package for Emacs. This is required for the xeft package to function.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix [hunk 1]
```diff
@@ -158,6 +158,23 @@ self: let
         }
       );
 
+      xeft = super.xeft.overrideAttrs (old: let
+        libExt = pkgs.stdenv.targetPlatform.extensions.sharedLibrary;
+      in {
+        dontUnpack = false;
+
+        buildInputs = (old.buildInputs or [ ]) ++ [pkgs.xapian];
+        buildPhase = ''
```
Comment (jian-lin) [inline pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix:167]: nitpick

```suggestion
        buildPhase = (old.buildPhase or "") + ''
```

**Review by jian-lin — COMMENTED** [review]
LGTM.

`xeft` is also available on [ELPA-devel](https://elpa.gnu.org/devel/xeft.html).  Could you make the same change for `pkgs/applications/editors/emacs/elisp-packages/elpa-devel-packages.nix`?

File: pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix (hunk 1 above)
Comment (acowley) [reply pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix]: Done.

**Comment by acowley** [comment]
Thanks for taking a look @jian-lin! I've added a commit that includes this change in `pkgs/applications/editors/emacs/elisp-packages/elpa-devel-packages.nix`.

**Review by jian-lin — APPROVED** [review]
Tested on Linux.

Could you change your commit message from `elpa-packages-devel: fix xeft build` to `elpa-devel-packages: build xeft dynamic module`?

File: pkgs/applications/editors/emacs/elisp-packages/elpa-devel-packages.nix [hunk 2]
```diff
@@ -63,6 +63,22 @@ self: let
           popd
         '';
       });
+
+      xeft = super.xeft.overrideAttrs (old: let
+        libExt = pkgs.stdenv.targetPlatform.extensions.sharedLibrary;
+      in {
+        dontUnpack = false;
+
+        buildInputs = (old.buildInputs or [ ]) ++ [pkgs.xapian];
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/elpa-devel-packages.nix:72]:
```suggestion
        buildInputs = (old.buildInputs or [ ]) ++ [ pkgs.xapian ];
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix [hunk 3]
```diff
@@ -158,6 +158,23 @@ self: let
         }
       );
 
+      xeft = super.xeft.overrideAttrs (old: let
+        libExt = pkgs.stdenv.targetPlatform.extensions.sharedLibrary;
+      in {
+        dontUnpack = false;
+
+        buildInputs = (old.buildInputs or [ ]) ++ [pkgs.xapian];
```
Comment (AndersonTorres) [inline pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix:166]:
```suggestion
        buildInputs = (old.buildInputs or [ ]) ++ [ pkgs.xapian ];
```

**Review by AndersonTorres — DISMISSED** [review]

File: pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix (hunk 3 above)
Comment (acowley) [reply pkgs/applications/editors/emacs/elisp-packages/elpa-packages.nix]: Done.

**Review by jian-lin — APPROVED** [review]

**Comment by AndersonTorres** [comment]
@ofborg build emacsPackages.xeft

**Pull request merged**